internal/ipc/internal/flisten/flisten.go

```diff
@@ -0,0 +1,60 @@
+package flisten
+
+import (
+	"context"
+	"errors"
+	"net"
+	"os"
+	"path/filepath"
+
+	"github.com/ActiveState/cli/internal/errs"
+	"github.com/ActiveState/cli/internal/ipc/namespace"
+)
+
+type FListen struct {
+	n *namespace.Namespace
+	net.Listener
+}
+
+func New(ctx context.Context, n *namespace.Namespace, network string) (*FListen, error) {
+	namespace := n.String()
+	namespaceDir := filepath.Dir(namespace)
```

Comment:
This feels error prone, `namespace` does not convey "filepath" to me. Suggest making that an explicit trait of the namespace or renaming namespace.